File: KotlinJavaNavegacion/app/src/main/java/com/example/kotlinjavanavegacion/MainActivity2.java
package com.example.kotlinjavanavegacion;

import android.content.Context;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity2 extends AppCompatActivity {

    Button btnBack;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);
        btnBack = findViewById(R.id.btnBack);
    }

    public void clickBack(View view){
        Context context = getApplicationContext();
        CharSequence text = "click boton back";
        int duration = Toast.LENGTH_SHORT;
        Toast toast = Toast.makeText(context, text, duration);
        toast.show();
        Intent downloadIntent = new Intent(this, MainActivity.class);
        startActivity(downloadIntent);
    }

}

File: KotlinJavaNavegacion/app/src/main/java/com/example/kotlinjavanavegacion/MainActivity2.kt
package com.example.kotlinjavanavegacion

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import com.example.kotlinjavanavegacion.R
import android.widget.Toast
import android.content.Intent
import android.view.View
import android.widget.Button
import com.example.kotlinjavanavegacion.MainActivity

class MainActivity2 : AppCompatActivity() {
    var btnBack: Button? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)
        btnBack = findViewById(R.id.btnBack)
    }

    fun clickBack(view: View?) {
        val context = applicationContext
        val text: CharSequence = "click boton back"
        val duration = Toast.LENGTH_SHORT
        val toast = Toast.makeText(context, text, duration)
        toast.show()
        val downloadIntent = Intent(this, MainActivity::class.java)
        startActivity(downloadIntent)
    }
}